src/Exception/InvalidArgumentException.php

```diff
@@ -1,14 +0,0 @@
-<?php
-
-namespace Http\Client\Exception;
-
-use Http\Client\Exception;
-
-/**
- * Custom Invalid Argument exception
- *
- * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
- */
-final class InvalidArgumentException extends \InvalidArgumentException implements Exception
```

Comment:
As said earlier I will only do that once the utility separation is ready to merge, so that it is clear what needs to go into the util repo.